openssl-sys/build/main.rs

```diff
@@ -115,6 +115,16 @@ fn main() {
         println!("cargo:rustc-link-lib={}={}", kind, lib);
     }
 
+    // https://github.com/openssl/openssl/pull/15086
+    if version == Version::Openssl3xx
+        && kind == "static"
+        && (env::var("CARGO_CFG_TARGET_OS").unwrap() == "linux"
+            || env::var("CARGO_CFG_TARGET_OS").unwrap() == "android")
+        && env::var("CARGO_CFG_TARGET_POINTER_WIDTH").unwrap() == "32"
+    {
+        println!("cargo:rustc-link-lib=dylib=atomic");
```

Comment:
When upgrading to 3.1, vendored tests fail with `undefined reference to `__atomic_stuff'` link errors, fixed by re-applying the patch removed in 32a303a752732fdd27c407a3a7fd668e546c05db. I did not dig the problem deeper.

Reply:
Hmm, I guess I'll need to poke around a bit to see if the commit that fixed this was reverted or something.

This is also something that may actually be best off moved to openssl-src.

Reply:
Ah, I think we just need to update the logic in openssl-src to include `atomic` as a lib when appropriate: https://github.com/alexcrichton/openssl-src-rs/blob/main/src/lib.rs#L549

Reply:
The libs to link don't seem to come from openssl-src (only the library path is [used](https://github.com/sfackler/rust-openssl/blob/master/openssl-sys/build/find_vendored.rs#L13)) but are [redefined in openssl-sys](https://github.com/sfackler/rust-openssl/blob/master/openssl-sys/build/main.rs#L109). We could try to take lib list from openssl-src but it would still require defining them for non-vendored cases so I'm not sure it would make sense.

Reply:
@sfackler We're new ~ 1 week from 1.1 EOL, is the current PR acceptable or do you want additional changes? As stated in the previous comment it does not seem practical to include the atomic lib on the openssl-src side.

Reply:
I Know this is merged already, but i wanted to provide my feedback.
This now broke armv6 `arm-unknown-linux-gnueabi` again, for which this was removed.
If this really is needed for vendored tests, then maybe only enable this on vendored tests, or, as i do this in my build scripts add it when needed like this `RUSTFLAGS="-Clink-args=-latomic"`.